cfme/common/vm.py

```diff
@@ -347,6 +368,44 @@ def get_detail(self, properties=None, icon_href=False):
         else:
             return InfoBlock.text(*properties)
 
+    def open_console(self, console='VM Console', invokes_alert=False, cancel=False):
+        """
+        Initiates the opening of one of the console types supported by the Access
+        button.   Presently we only support VM Console, which is the HTML5 Console.
+        In case of VMware provider it could be VMRC, VNC/HTML5, WebMKS, but we only
+        support VNC/HTML5.
+        Possible values for 'console' could be 'VM Console' and 'Web Console', but Web
+        Console is not supported as well.
+
+        Args:
+            console       - one of the supported console types given by the Access button.
+            invokes_alert - If the particular console will invoke a CFME popup/alert
+                            setting this to true will handle this.
+            cancel        - Allows one to cancel the operation if the popup/alert occurs.
+        """
+        # TODO: implement vmrc vm console
+        if console not in ['VM Console']:
+            raise NotImplementedError('Not supported console type: {}'.format(console))
+
+        view = navigate_to(self, 'Details')
+
+        # Click console button given by type
+        view.toolbar.access.item_select(console)
+
+        # Get the consoles window handle, and then create a VMConsole object, and store
+        # the VMConsole object aside.
+        console_handle = self.console_handle
```

Comment:
**Required**
Basic handling of when console_handle returns none here. raise an exception.